llvm/test/CodeGen/NVPTX/ldu-ldg.ll

```diff
@@ -148,3 +148,129 @@ define <2 x half> @test_ldg_v2f16(ptr addrspace(1) %ptr) {
   %val = tail call <2 x half> @llvm.nvvm.ldg.global.f.v2f16.p1(ptr addrspace(1) %ptr, i32 4)
   ret <2 x half> %val
 }
+
+; CHECK-LABEL: test_ldu_i8_dead
+define void @test_ldu_i8_dead(ptr addrspace(1) %ptr) {
+  ; CHECK-NOT: ldu.global.u8
```

Comment:
-NOT checks are fragile. Can you precommit generated checks? Also, why would deleting these dead intrinsics need to be done in codegen and not already done in the middle end?

Reply:
Previously these weren't getting deleted before codegen since they weren't `willReturn`. With the added `IntrWillReturn` attribute they can be deleted in the middle end, so I'll get rid of these tests and make ones to reflect that.